ludwig/datasets/loaders/dataset_loader.py

```diff
@@ -451,6 +451,28 @@ def load_unprocessed_dataframe(self, file_paths: List[str]) -> pd.DataFrame:
             _list_of_strings(self.config.validation_filenames), root_dir=self.raw_dataset_dir
         )
         test_paths = _glob_multiple(_list_of_strings(self.config.test_filenames), root_dir=self.raw_dataset_dir)
+        dataframes = []
+        if self.config.huggingface_dataset_id:
+            dataframes = self._get_dataframe_with_fixed_splits_from_hf()
+        else:
+            dataframes = self._get_dataframe_with_fixed_splits(
+                train_paths, validation_paths, test_paths, dataset_paths, file_paths
+            )
+        return pd.concat(dataframes, ignore_index=True)
+
+    def _get_dataframe_with_fixed_splits_from_hf(self):
+        dataframes = []
+        splits = ["train", "validation", "test"]
+        data_dict = self.load_hf_to_dataframe(self.config.huggingface_dataset_id, self.config.huggingface_subset)
```

Comment:
Is this new? Why do we need to modify `load_unprocessed_dataframe`? Where is `load_hf_to_dataframe` defined?

Reply:
load_hf_to_dataframe is defined in `ludwig/datasets/loaders/hugging_face.py`

Reply:
We modify `load_unprocessed_dataframe` in order to handle how HF datasets are loaded in